inject.js

```diff
@@ -56,6 +56,18 @@ chrome.extension.sendMessage({}, function(response) {
       }
       this.initializeControls();
 
+      //Save the speed of current tab's video before leaving the page
+      //(ie. reloading page or going Backward/Forward in history)
+      //so that it doesn't restore the speed set by a video in another tab
+      //but instead, keeps the current tab's video speed the same,
+      //until changed by user or browser quits.
+      window.onbeforeunload = function(e) {
```

Comment:
A new tab gets the last seen/fiddled speed, which is either the speed of the last closed/reloaded/Back/Forwarded video tab, or the speed that was last changed eg. via A/D keys, whichever was last.
It seems kinda normal to me. Unless I'm missing something? :)

The only problem is: you don't know which speed will be saved when closing the whole browser or window(with a bunch of video tabs); tab visibility order, I guess, would be deciding which speed was last and that one will be remembered/restored when browser is started again or new video tab opened.


Previously(currently), the speed of the new tab(and on browser restart) was the speed that was last changed eg. via A/D keys.



But, you don't have to consider merging this; I just wanted it to be available here in PRs for anyone else (like me) who needed this behavior, if any.